lib/mix/tasks/compile.make.ex

```diff
@@ -106,7 +111,8 @@ defmodule Mix.Tasks.Compile.ElixirMake do
   end
 
   defp build(config, task_args) do
-    exec      = Keyword.get(config, :make_executable, :default) |> os_specific_executable()
+    exec      = System.get_env("MAKE") |> sanitize_input() ||
```

Comment:
Maybe let's turn this into

```elixir
exec = executable_from_enviroment_variable() || executable_from_config(config)
```

so that we don't have all this logic in just an assignment expression (which also spans two lines). Wdyt